src/lib.rs

```diff
@@ -644,16 +644,31 @@ pub type Result<T> = std::result::Result<T, Status>;
 
 ////////////////////////
 
+/// A common implementation of the sealed supertrait programming pattern 
+/// (C-SEALED)
+/// 
+/// With this, implementations of `Sealed` are guarenteed to only exist in the
+/// current crate. This allows modifictation of the underyling traits in ways
+/// that would ordinarily be a breaking change for traits that are not sealed.
+/// 
+/// See https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed
+ 
+mod private {
+    pub trait Sealed {}
+    
+    impl <T> Sealed for T{}
```

Comment:
I don't think this blanket impl makes sense, because it implements Sealed for all types, even for types defined in other crates. That makes the `TensorType: private::Sealed` bound useless, because it's implemented for every type already.

Reply:
Yep, please fix.